Mathlib/FieldTheory/SeparableClosure.lean

```diff
@@ -0,0 +1,386 @@
+  exact ⟨y, hx, hy⟩
+
+/-- If `i` is an `F`-algebra isomorphism of `E` and `K`, then `separableClosure F K` is equal to
+the image of `separableClosure F E` under the map `i`. -/
+theorem separableClosure.eq_map_of_algEquiv (i : E ≃ₐ[F] K) :
+    separableClosure F K = (separableClosure F E).map i.toAlgHom := by
+  refine le_antisymm (fun x hx ↦ ?_) (map_le_of_algHom F E K i.toAlgHom)
+  change x ∈ (_ : Set K)
+  rw [coe_map, Set.mem_image]
+  exact ⟨i.symm.toAlgHom x,
+    (map_mem_separableClosure_iff F K E i.symm.toAlgHom).2 hx, i.apply_symm_apply x⟩
+
+/-- If `E` and `K` are isomorphic as `F`-algebras, then `separableClosure F E` and
+`separableClosure F K` are also isomorphic as `F`-algebras. -/
+def separableClosure.algEquivOfAlgEquiv (i : E ≃ₐ[F] K) :
+    separableClosure F E ≃ₐ[F] separableClosure F K :=
+  ((separableClosure F E).intermediateFieldMap i).trans
+    (equivOfEq (eq_map_of_algEquiv F E K i).symm)
+
+/-- The (relative) separable closure of `E / F` is algebraic over `F`. -/
+theorem separableClosure.isAlgebraic : Algebra.IsAlgebraic F (separableClosure F E) :=
+  fun x ↦ isAlgebraic_iff.2 x.2.isIntegral.isAlgebraic
+
+/-- The (relative) separable closure of `E / F` is separable over `F`. -/
+instance separableClosure.isSeparable : IsSeparable F (separableClosure F E) :=
+  ⟨fun x ↦ by simpa only [minpoly_eq] using x.2⟩
+
+/-- An intermediate field of `E / F` is contained in the (relative) separable closure of `E / F`
+if all of its elements are separable over `F`. -/
+theorem le_separableClosure' {L : IntermediateField F E} (hs : ∀ x : L, (minpoly F x).Separable) :
+    L ≤ separableClosure F E := fun x h ↦ by simpa only [minpoly_eq] using hs ⟨x, h⟩
+
+/-- An intermediate field of `E / F` is contained in the (relative) separable closure of `E / F`
+if it is separable over `F`. -/
+theorem le_separableClosure (L : IntermediateField F E) [IsSeparable F L] :
+    L ≤ separableClosure F E := le_separableClosure' F E (IsSeparable.separable F)
+
+/-- An intermediate field of `E / F` is contained in the (relative) separable closure of `E / F`
+if and only if it is separable over `F`. -/
+theorem le_separableClosure_iff (L : IntermediateField F E) :
+    L ≤ separableClosure F E ↔ IsSeparable F L :=
+  ⟨fun h ↦ (isSeparable_def _ _).2 fun x ↦ by simpa only [minpoly_eq] using h x.2,
+    fun _ ↦ le_separableClosure _ _ _⟩
+
+/-- The (relative) separable closure of the (relative) separable closure of `E / F` is equal to
+itself. -/
+theorem separableClosure.separableClosure_eq_bot :
+    separableClosure (separableClosure F E) E = ⊥ := bot_unique fun x hx ↦ by
+  rw [mem_separableClosure_iff, ← isSeparable_adjoin_simple_iff_separable] at hx
+  set L := separableClosure F E
+  haveI : IsSeparable F (L⟮x⟯.restrictScalars F) := IsSeparable.trans F L L⟮x⟯
+  exact mem_bot.2 ⟨⟨x, (mem_separableClosure_iff F E).2 (separable_of_mem_isSeparable F E <|
+    show x ∈ L⟮x⟯.restrictScalars F from mem_adjoin_simple_self L x)⟩, rfl⟩
```

Comment:
I think we should extract from the proof of `IsSeparable.trans` the lemma that `(minpoly L x).Separable` and `IsSeparable F L` implies `(minpoly F x).Separable`. That would also make the proof of `eq_restrictScalars_of_isSeparable` just one line.